ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js

```diff
@@ -475,11 +475,86 @@ module.exports = function CesiumMap(
+      const ellipsoidGeodesic = new Cesium.EllipsoidGeodesic(startCartographic, endCartographic)
+      
+      return ellipsoidGeodesic.surfaceDistance
+    },
+    /*
+     * Draws a marker on the map designating a start/end point for the ruler measurement. The given
+     * coordinates should be an object with 'lat' and 'lon' keys with degrees values. The given
+     * marker label should be a single character or digit that is displayed on the map marker.
+     */
+    addRulerPoint(coordinates, markerLabel) {
+      const { lat, lon } = coordinates
+      // a point requires an altitude value so just use 0
+      const point = [lon, lat, 0]
+      const options = {
+        id: markerLabel,
+        title: `Selected ruler coordinate '${markerLabel}'`,
+        color: '#FC2803',
+        icon: null,
+        view: this,
+      }
+
+      // calls addPointWithText() with the flag set to use the custom marker label
+      return this.addPointWithText(point, options, true)
+    },
+    /*
+     * Removes the given Billboard from the map.
+     */
+    removeRulerPoint(billboardRef) {
+      billboardCollection.remove(billboardRef)
+    },
+    /*
+     * Draws a line on the map between the points in the given array of Billboards.
+     */
+    addRulerLine(billboards) {
+      // creates an array of Cartesian3 points
+      const cartesianArray = billboards.map(billboard => billboard.position)
+      // a PolylineGeometry allows the line to follow the curvature of the surface
+      const polyline = new Cesium.PolylineGeometry({
+        positions: cartesianArray,
+        width: 8,
+      })
+      const geometry = Cesium.PolylineGeometry.createGeometry(polyline)
+      const geometryInstance = new Cesium.GeometryInstance({
+        geometry: geometry,
+        attributes: {
+          color: Cesium.ColorGeometryInstanceAttribute.fromColor(Cesium.Color.DEEPPINK)
+        }
+      })
+      const linePrimitive = new Cesium.Primitive({
+        geometryInstances: geometryInstance,
+        appearance: new Cesium.PolylineColorAppearance()
+      })
+      // calculates the distance between the two positions in the array of Cartesian3 points
+      const distanceBetweenPoints = this.calculateDistanceBetweenPositions(cartesianArray)
+      mapModel.setCurrentDistance(distanceBetweenPoints)
+      // rendering the line breaks if the distance is 0
+      if (distanceBetweenPoints > 0) {
+        map.scene.primitives.add(linePrimitive)
+      }
+
+      return linePrimitive
+    },
+    /*
+     * Removes the given line Primitive from the map.
+     */
+    removeRulerLine(primitive) {
+      map.scene.primitives.remove(primitive)
+    },
     /*
             Adds a billboard point utilizing the passed in point and options.
             Options are a view to relate to, and an id, and a color.
           */
-    addPointWithText(point, options) {
+    addPointWithText(point, options, useCustomText=false) {
```

Comment:
Added a default parameter here to be able to reuse this method in `addRulerPoint()` with a custom text while not breaking any previous uses.